### PR TITLE
add bool_decisions to stats

### PR DESCRIPTION
**Pull request by louisabraham**
Useful to monitor the performance in some cases.

### CODE REVIEW

File: src/core/Internal.ml [hunk 1]
```diff
@@ -143,6 +143,7 @@ type t = {
 
   mutable starts : int; (* number of (re)starts *)
   mutable decisions : int; (* number of decisions *)
+  mutable bool_decisions : int; (* number of decisions *)
```
Comment (c-cube) [inline src/core/Internal.ml]: the comment should read "number of boolean decision" :-)

File: src/core/Internal.ml [hunk 2]
```diff
@@ -450,7 +456,7 @@ let partition_atoms (atoms:atom array) : atom list * raw_premise_step list =
           raise Trivial (* A var true at level 0 gives a trivially true clause *)
         ) else (
           (a :: trues) @ unassigned @ falses @
-            (arr_to_list atoms (i + 1)), history
+          (arr_to_list atoms (i + 1)), history
```
Comment (c-cube) [inline src/core/Internal.ml:459]: why the reindent?

File: src/core/Internal.ml [hunk 3]
```diff
@@ -726,7 +732,7 @@ let eval_atom_to_false ~save (env:t) (a:atom): unit =
   ) else (
     let v = Atom.eval a in
     Log.debugf debug (fun k->k "(@[atom_must_be_false@ %a@ :eval_to %a@])"
-        Atom.debug a debug_eval v);
+                         Atom.debug a debug_eval v);
```
Comment (c-cube) [inline src/core/Internal.ml:735]: ditto

File: src/core/Internal.ml [hunk 4]
```diff
@@ -1468,7 +1475,7 @@ let rec gc_mark_clause (c:clause) : unit =
   if not (Clause.gc_marked c) then (
     Log.debugf 15 (fun k->k "(@[gc_mark_clause@ %a@])" Clause.pp_name c);
     Clause.gc_mark c;
-    Array.iter (gc_mark_atom ~mark_clause:true) c.c_atoms;
+    Array.iter (gc_mark_atom ~mark_clause:true) c.c_atoms
```
Comment (c-cube) [inline src/core/Internal.ml]: unrelated

**Review by c-cube — CHANGES_REQUESTED** [review]
good, but some unrelated changes sneaked in.

File: src/core/Internal.ml (hunk 2 above)
Comment (louisabraham) [reply src/core/Internal.ml:459]: ocp-indent did it alone :-)

**Comment by louisabraham** [comment]
Sorry, I use ocp-indent to auto-indent. It should be okay now.

**Comment by c-cube** [comment]
There are still lots of indentation changes… It's fine to reindent, but not mixed with other things

**Comment by louisabraham** [comment]
Sorry, didn't know this "rule".
Is it OK to do the indentation in a separate commit but in the same PR?

File: src/core/Internal.ml [hunk 5]
```diff
@@ -1712,24 +1719,24 @@ let solve
...
-          );
-
-          (* increment parameters to ensure termination *)
-          n_of_conflicts := !n_of_conflicts *. env.restart_inc;
-          n_of_learnts   := !n_of_learnts *. env.learntsize_inc;
-          (* diminish by how much n_of_learnts increases *)
-          env.learntsize_inc <- 1. +. (env.learntsize_inc -. 1.) /. 1.3 ;
-          loop()
-        | exception Sat -> check_sat ()
+      | () -> ()
```
Comment (c-cube) [inline src/core/Internal.ml:1722]: is this really produced by ocp-indent‽ I thought the config required 2 spaces…

**Comment by c-cube** [comment]
Yeah, putting it in a separate commit is fine. I'll rune `make reindent` afterwards anyway.

File: src/core/Internal.ml (hunk 5 above)
Comment (louisabraham) [reply src/core/Internal.ml:1722]: Yes it is. 
I think it is logical: the `begin` produces a new block, but the match doesn't.
You cannot indent two times.

Comment (louisabraham) [reply src/core/Internal.ml:1722]: Thanks for making me discover the interrobang 💯 💯 💯

**Comment by louisabraham** [comment]
I am a bad person who does 'git push -f' and I know it :)

It should be OK now.

**Comment by c-cube** [comment]
`git push -f` is fine if you do it on your own branches :)
